### PR TITLE
Dynamic Nation Colors

### PR DESCRIPTION
**Pull request by Goosius1**
**SUMMARY**
- This is a feature for dynamic nation colours
- The feature works in combination with new code in the Towny-Dynmap plugin.
- The towny-dynmap code can be reviewed in the "Dynamic nation colours" PR in the towny-dynmap repo.

**DESCRIPTION**
- This feature immediately gives colors to all nations on the dynmap when the updated Towny and Dynmap towny jars are deployed
- After it does this, it allows the colors to be changed easily, without restarting the dynmap or towny plugins
- An allowed list of colors is found in the Towny plugin config.yml
- Existing nations automatically get a random allowed color when the server is first restarted
- New nations automatically get a random allowed color
- The command to change color is '/n set mapcolor [color]'
- By default kings can change their nation color, but this can be restricted by permission
- Admins can change nation colors using 'ta nation [nation] set mapcolor [color]'

____
#### New Nodes/Commands/ConfigOptions: 
- New config option in dynmap:  "dynamic-nation-colors"
- New permission node in towny:  towny.command.nation.set.mapcolor
- New command "/n set mapcolor [color]"

____
- [x] I have tested this pull request for defects on a server. 

By making this pull request, I represent that I have the right to waive copyright and related rights to my contribution, and agree that all copyright and related rights in my contributions are waived, and I acknowledge that the TownyAdvanced organization has the copyright to use and modify my contribution under the Towny [License](https://github.com/LlmDl/Towny/blob/master/LICENSE.md) for perpetuity.

### CODE REVIEW

File: src/com/palmergames/bukkit/towny/permissions/PermissionNodes.java [hunk 1]
```diff
@@ -50,6 +50,7 @@
 		TOWNY_COMMAND_NATION_SET_BOARD("towny.command.nation.set.board"),
 		TOWNY_COMMAND_NATION_SET_SPAWN("towny.command.nation.set.spawn"),
 		TOWNY_COMMAND_NATION_SET_SPAWNCOST("towny.command.nation.set.spawncost"),
+		TOWNY_COMMAND_NATION_SET_MAPCOLOR("towny.command.nation.set.mapcolor"),
```
[the remark this answers is not in the document]
Comment (Goosius1) [reply src/com/palmergames/bukkit/towny/permissions/PermissionNodes.java:53]: Done

File: resources/english.yml [hunk 2]
```diff
@@ -1102,4 +1102,6 @@ msg_reloaded_config: '&2Configuration successfully reloaded.'
 msg_reloaded_perms: '&2Permissions successfully reloaded.'
 msg_reloaded_db: '&2Database successfully reloaded'
 msg_reloaded_lang: '&2Language file successfully reloaded'
-msg_reset_config: '&2Config has been reset'
\ No newline at end of file
+msg_reset_config: '&2Config has been reset'
+msg_err_invalid_nation_map_color: '&cUnknown map color. Allowed colors are %s.'
+msg_nation_map_color_changed: '&bThe nation map color was changed to %s.'
```
Comment (suneettipirneni) [inline resources/english.yml]: You've updated the English the language file, however, I don't see any updates for the other language files. Also, the version number on each language file needs to be bumped anytime there is an addition.

Comment (LlmDl) [reply resources/english.yml]: > You've updated the English the language file, however, I don't see any updates for the other language files. Also, the version number on each language file needs to be bumped anytime there is an addition.

Not updating all the lang files is fine, I can do any of that later on. Having them all in place this early on can end up making lang file conflicts more difficult.

Comment (Goosius1) [reply resources/english.yml]: * Bumped english lang file version
* Added comment attributing the 2 new lines to the new version

**Comment by LlmDl** [comment]
This just has to have its english.yml fixed and I'll get this into 0.96.1.3 as we're already having to bump the language files.

**Comment by Goosius1** [comment]
Great!.
Should I bump the version to 80 in this PR,  or should the changes be under 79?

**Comment by LlmDl** [comment]
Nope it'll be part of 0.79

**Comment by Goosius1** [comment]
Code conflict resolved

**Pull request merged**